spec/formatting.md

```diff
@@ -307,13 +307,13 @@ the following steps are taken:
    resolve the value of the _expression_ as the result of that function call.
 
    If the call fails or does not return a valid value,
-   emit a _Invalid Expression_ error.
+   emit a _Message Function Error_.
```

Comment:
That seems a bit too opinionated? As in, an implementation could have a hierarchy of errors like Error > MessageFormatError > MessageFunctionError > BadOption, and have a need to use the MessageFunctionError class directly if user code throws a random error that's not a MessageFunctionError.

Reply:
... in which case the "appropriate" error is the parent class of error. My suggestion permits the implementation to emit one of the subclasses of error and encourages the implementer to figure out which one it is. Generally speaking, if there's a random exception thrown, it'll be emitted as a FormattingError or SelectionError (indicating what the implementation was trying to do when it failed). This would even be helpful in debugging for functions that do both selection and formatting, since it would tell me which part of the function the error took place in.